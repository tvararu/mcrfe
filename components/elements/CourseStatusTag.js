const enrichmentHasUnpublishedChanges = enrichment => {
  return (
    enrichment &&
    enrichment.status === 0 &&
    enrichment.lastPublishedTimestampUtc
  );
};

const enrichmentContentStatusContent = enrichment => {
  if (!enrichment) return "Empty";
  if (enrichment.status === 1) return "Published";
  if (enrichmentHasUnpublishedChanges(enrichment)) return <>Published&nbsp;*</>;
  return "Draft";
};

const enrichmentContentStatusCss = enrichment => {
  if (!enrichment) return "empty";
  if (enrichment.status === 1 || enrichment.lastPublishedTimestampUtc)
    return "published";
  return "draft";
};

export default ({ enrichment }) => (
  <>
    <style jsx>{`
      .div {
        padding: 4px 8px 1px;
      }

      .empty {
        background: #ffffff;
        color: #6f777b;
        border: 2px solid #899094;
        margin-top: -1px;
        margin-bottom: -1px;
      }

      .draft {
        background: #f47738;
      }

      .published {
        background: #00823b;
      }
    `}</style>
    <div
      className={`govuk-tag tag--small ${enrichmentContentStatusCss(
        enrichment
      )}`}
    >
      {enrichmentContentStatusContent(enrichment)}
    </div>
    {enrichmentHasUnpublishedChanges(enrichment) && (
      <div className="govuk-body govuk-body-s govuk-!-margin-bottom-0 govuk-!-margin-top-1">
        *&nbsp;Unpublished&nbsp;changes
      </div>
    )}
  </>
);
