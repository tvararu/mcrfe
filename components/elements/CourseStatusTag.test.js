import renderer from "react-test-renderer";
import CourseStatusTag from "./CourseStatusTag";

const enrichments = [
  null,
  {
    _label: "draft",
    status: 0,
    lastPublishedTimestampUtc: null
  },
  {
    _label: "published*",
    status: 0,
    lastPublishedTimestampUtc: 1
  },
  {
    _label: "published",
    status: 1,
    lastPublishedTimestampUtc: null
  }
];

describe("CourseStatusTag", () => {
  enrichments.forEach(enrichment => {
    it(`renders correctly with ${(enrichment && enrichment._label) ||
      "empty"} enrichment`, () => {
      const component = renderer.create(
        <CourseStatusTag enrichment={enrichment} />
      );
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
