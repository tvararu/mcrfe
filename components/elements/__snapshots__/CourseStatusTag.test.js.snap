// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseStatusTag renders correctly with draft enrichment 1`] = `
<div
  className="jsx-4100856793 govuk-tag tag--small draft"
>
  Draft
</div>
`;

exports[`CourseStatusTag renders correctly with empty enrichment 1`] = `
<div
  className="jsx-4100856793 govuk-tag tag--small empty"
>
  Empty
</div>
`;

exports[`CourseStatusTag renders correctly with published enrichment 1`] = `
<div
  className="jsx-4100856793 govuk-tag tag--small published"
>
  Published
</div>
`;

exports[`CourseStatusTag renders correctly with published* enrichment 1`] = `
Array [
  <div
    className="jsx-4100856793 govuk-tag tag--small published"
  >
    Published *
  </div>,
  <div
    className="jsx-4100856793 govuk-body govuk-body-s govuk-!-margin-bottom-0 govuk-!-margin-top-1"
  >
    * Unpublished changes
  </div>,
]
`;
